src/pages/home-page/home-page.tsx

```diff
@@ -1,18 +1,60 @@
-import React from "react";
-import { Header } from "../../common/components/header/header";
+import React, { useEffect, useState } from "react";
 import { NoQuizzes } from "../no-quizzes-page/no-quizzes";
 import { QuizList } from "../quiz-list-page/quiz-list-page";
 
 import "../../styles/buttons.scss";
+import "./home-page.scss";
+import { Layout } from "../layout-page/layout";
+import { QuizStore } from "../../features/create-quiz/store/CreateQuizStore";
+import { toJS } from "mobx";
+import { collection, getDocs, query, where } from "firebase/firestore";
+import { db } from "../../config/firebase-config";
+import { LoadingScreen } from "../../common/components/loading-screen/loading-screen";
+import { LoginStore } from "../../features/authentication/login/store/LoginStore";
 
 interface HomePageProps {}
 
-export const HomePage: React.FC<HomePageProps> = ({}) => {
+export const HomePage: React.FC<HomePageProps> = () => {
+  const [quizzes, setQuizzes] = useState<{ [key: string]: any }[]>([]);
+  const quizzesCollectionRef = collection(db, "quizzes");
+
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    setTimeout(() => {
+      setLoading(false);
+    }, 3000);
```

Comment:
Using `setTimeout` is not a reliable method. I suggest setting the `loading` state to `false` at the end of the `getQuizzes` function execution. This way you know for sure when the processing is done.